src/main/java/org/lantern/http/InteractionServlet.java

```diff
@@ -171,11 +173,13 @@ protected void processRequest(final HttpServletRequest req,
             return;
         }
 
-        if (!SecurityUtils.constantTimeEquals(model.getXsrfToken(),
-                req.getHeader("X-XSRF-TOKEN"))) {
-            log.debug("X-XSRF-TOKEN wrong: got {} expected {}", req.getHeader("X-XSRF-TOKEN"), model.getXsrfToken());
-            HttpUtils.sendClientError(resp, "invalid X-XSRF-TOKEN");
-            return;
+        if (!LanternUtils.isDevMode() && !LanternUtils.isLanternPi()) {
+            if (!SecurityUtils.constantTimeEquals(model.getXsrfToken(),
+                    req.getHeader("X-XSRF-TOKEN"))) {
+                log.debug("X-XSRF-TOKEN wrong: got {} expected {}", req.getHeader("X-XSRF-TOKEN"), model.getXsrfToken());
+                HttpUtils.sendClientError(resp, "invalid X-XSRF-TOKEN");
+                return;
+            }
         }
```

Comment:
Sorry, I don't get it.  Public sites use XSRF protection all the time.  What exactly is going wrong with XSRF on?